### PR TITLE
Fix for issue 333

### LINKED ISSUE #333: Wrong D and F MO coefficient order in QUICK generated .molden files
The MO coefficient section in .molden files written by existing molden exporter in QUICK doesn't follow the correct format for D and F orbitals. The correct format is as follows (ref:  https://www.theochem.ru.nl/molden/molden_format.html).  

```
   5D: D 0, D+1, D-1, D+2, D-2
   6D: xx, yy, zz, xy, xz, yz

   7F: F 0, F+1, F-1, F+2, F-2, F+3, F-3
  10F: xxx, yyy, zzz, xyy, xxy, xxz, xzz, yzz, yyz, xyz
```
This must be fixed ASAP.

### PR DESCRIPTION
**Pull request by Madu86**
The enclosed changes resolve #333. Please check and merge.

### CODE REVIEW

File: src/modules/quick_molden_module.f90 [hunk 1]
```diff
@@ -97,11 +100,11 @@ subroutine write_coordinates(self, ierr)
            ! we need to do this because optimizers may return the geometry
            ! for the next step which may be stored in xyz
            k = self%iexport_snapshot - 1
-           write(self%iMoldenFile, '(3(4x,F10.4))') (self%xyz_snapshots(j,i,k),j=1,3)
+           write(self%iMoldenFile, '(3(4x,F13.10))') (self%xyz_snapshots(j,i,k),j=1,3)
```
Comment (agoetz) [inline src/modules/quick_molden_module.f90:103]: This looks dangerous. F13.10 has field width 13 and 10 significant digits. Add one for the decimal point and one for  a minus sign, and this leaves only a single digit for the integer part of the coordinate.

I would use '3(2x,F20.10))' as format.

File: src/modules/quick_molden_module.f90 [hunk 2]
```diff
@@ -97,11 +100,11 @@ subroutine write_coordinates(self, ierr)
...
          else
            ! if it's a single point calculation we can use xyz
            ! we can't use xyz_snapshots because they have not been populated
-           write(self%iMoldenFile, '(3(4x,F10.4))') (xyz(j,i),j=1,3)
+           write(self%iMoldenFile, '(3(4x,F13.10))') (xyz(j,i),j=1,3)
```
Comment (agoetz) [inline src/modules/quick_molden_module.f90:107]: Same as above suggestion to use '3(2x,F20.10))' format.

File: src/modules/quick_molden_module.f90 [hunk 3]
```diff
@@ -189,16 +229,19 @@ subroutine write_mo(self, ierr)
         endif
 
         write(lbl1,'(I5)') i
-        write(self%iMoldenFile, '(A11)') "Sym= a"//trim(adjustl(lbl1))
-        write(self%iMoldenFile, '(2x, "Ene= ", E16.10)') quick_qm_struct%E(i)
+        write(self%iMoldenFile, '(A11)') "  Sym= a"//trim(adjustl(lbl1))
+        write(self%iMoldenFile, '("  Ene= ", F15.10)') quick_qm_struct%E(i)
```
Comment (agoetz) [inline src/modules/quick_molden_module.f90:233]: This does not allow for large energies. Use F20.20 or scientific notation.

File: src/modules/quick_molden_module.f90 [hunk 4]
```diff
@@ -212,20 +255,25 @@ subroutine write_mo(self, ierr)
             endif
     
             write(lbl1,'(I5)') i
-            write(self%iMoldenFile, '(A11)') "Sym= b"//trim(adjustl(lbl1))
-            write(self%iMoldenFile, '(2x, "Ene= ", E16.10)') quick_qm_struct%Eb(i)
+            write(self%iMoldenFile, '(A11)') "  Sym= b"//trim(adjustl(lbl1))
+            write(self%iMoldenFile, '("  Ene= ",F15.10)') quick_qm_struct%Eb(i)
```
Comment (agoetz) [inline src/modules/quick_molden_module.f90:259]: Same as above, allow larger total energy numbers (use F20.20 or scientific notation).

**Review by agoetz — CHANGES_REQUESTED** [review]
Format specifiers need adjusting to allow larger numbers for coordinates and total energy,

**Comment by Madu86** [comment]
@agoetz, Thanks for pointing these out. I have made changes. Please check.

**Pull request merged**